lib/system/threads.nim

```diff
@@ -185,6 +185,14 @@ else:
         abi: array[56 div sizeof(clong), clong]
       ThreadVarSlot {.importc: "pthread_key_t",
                     header: "<sys/types.h>".} = distinct cuint
+  elif defined(nintendoswitch):
+    type
+      SysThread* {.importc: "pthread_t",
+                  header: "<sys/types.h>" .} = distinct cuint
+      Pthread_attr {.importc: "pthread_attr_t",
+                    header: "<sys/types.h>".} = object
+      ThreadVarSlot {.importc: "pthread_key_t",
+                    header: "<sys/types.h>".} = distinct cuint
```

Comment:
This code is very similar to the code above it, can the above code be reused for this branch?

Reply:
Yeah, you're right. I just wasn't sure about the `abi: array[56 div sizeof(clong), clong]`, but it looks like it's fine.